x/emissions/module/rewards/scores.go

```diff
@@ -111,19 +111,26 @@ func GenerateReputerScores(
 		newScores = append(newScores, newScore)
 	}
 
+	types.EmitNewReputerScoresSetEvent(ctx, newScores)
```

Comment:
the scores aren't necessarily set here though right? Do we want to emit somewhere in the calling function?

Reply:
they're set just before. The event should be emitted when it was set under certain conditions, which I'd say is responsibility for context above. ofc we don't use the set function elsewhere... but events seem more tied to reward imo